anyway/infographics_utils.py

```diff
@@ -1118,91 +1121,67 @@ def localize_items(request_params: RequestParams, items: Dict) -> Dict:
 
 
 @register
-class InjuredAccidentsWithPedestriansWidget(SubUrbanWidget):
+class InjuredAccidentsWithPedestriansWidget(UrbanWidget):
     name: str = "injured_accidents_with_pedestrians"
 
+    def validate_parameters(self):
+        # TODO: validate each parameter and display message accordingly
+        return self.request_params.location_info.get('yishuv_name') is not None and \
+            self.request_params.news_flash_obj.street1_hebrew is not None and  \
+            self.request_params.years_ago is not None
+
+    def convert_to_dict(self, query_results):
+        res = {}
+
+        for query_result in query_results:
+            if query_result.injury_severity not in res:
+                res[query_result.injury_severity] = {}
+            if query_result.accident_year not in res[query_result.injury_severity]:
+                res[query_result.injury_severity][query_result.accident_year] = 0
+
+            res[query_result.injury_severity][query_result.accident_year] += query_result.count
+
+        return res
+
     def __init__(self, request_params: RequestParams):
         super().__init__(request_params, type(self).name)
         self.rank = 18
-        self.text = {"title": "נפגעים הולכי רגל ברחוב ז׳בוטינסקי, פתח תקווה"}
-
-    @staticmethod
-    def is_in_cache() -> bool:
-        return False
 
     def generate_items(self) -> None:
-        self.items = (
-            InjuredAccidentsWithPedestriansWidget.injured_accidents_with_pedestrians_mock_data()
-        )
+        try:
+            if not self.validate_parameters():
+                logging.exception(f"Could not validate parameters for {NewsFlash} : {self.request_params.news_flash_obj.id}")
+                return None
+
+            query = db.session.query(InvolvedMarkerView)    \
+                .with_entities(InvolvedMarkerView.accident_year,    \
+                    InvolvedMarkerView.injury_severity, \
+                    func.count().label('count'))  \
+                .filter(InvolvedMarkerView.accident_yishuv_name == self.request_params.location_info['yishuv_name'])\
+                .filter(InvolvedMarkerView.injury_severity.in_([InjurySeverity.KILLED.value, InjurySeverity.SEVERE_INJURED.value, InjurySeverity.LIGHT_INJURED.value]))   \
+                .filter(InvolvedMarkerView.injured_type == InjuredType.PEDESTRIAN.value)   \
+                .filter(or_(InvolvedMarkerView.street1_hebrew == self.request_params.news_flash_obj.street1_hebrew, InvolvedMarkerView.street2_hebrew == self.request_params.news_flash_obj.street1_hebrew))    \
```

Comment:
The location information should be taken from `self.request_params`, and not directly from `self.news_flash_obj`, due to two reasons:
1.  We apply some logic when setting the location from the `NewsFlash` into `request_params`.
2.  We are starting to generate infographics from location only, so we will not have `news_flash_obj`